database/database.js

```diff
@@ -0,0 +1,45 @@
+const mongoose = require('mongoose');
+const db = mongoose.connection;
+
+mongoose.connect('mongodb://localhost/fec_product_features', {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useFindAndModify: false,
+  useCreateIndex: true
+});
+
+db.on('error', console.error.bind(console, 'connection error:'));
+db.once('open', function() {
+  console.log('Mongoose connected established.');
+});
+
+const productFeaturesSchema = new mongoose.Schema({
+  productId: {
+    type: Number,
+    unique: true
+  },
+  banner: {
+    header: String,
+    text: String
+  },
+  features: [{
+    header: String,
+    description: String
+  }],
+  featureSetup: {
+    header: String,
+    description: []
+  },
+  additionalFeatures: {
+    header: String,
+    description: String,
+    contentGrid: [{
+      title: String,
```

Comment:
We have objects and each key also has value as object with multiple key value pairs. If you need to do data updates ( not applicable for current scope where we are just reading) , would having all properties at same level be easier ?

Reply:
Hi Barsha, thank you for doing a code review. I appreciate all of your comments and questions! 

I made the decision to have a nested data structure because of the repetitive nature of the product-features service. Each document will have 4 primary sections (`banner`, `features`, `featureSetup`, and `additionalFeatures`), each containing their own header and  description value. The `features` key is an array of objects, which alone will hold 7 objects each with a unique `header` and `description` value. Having a structure on the same level would force the schema to use unique keys for each `feature` paragraph. It is in my opinion easier to have a nested structure that can be mapped over when rendering on the client.

While having a structure with data at the same level may be easier to read, I do not believe it would work well with the product-features service. Lastly, I will need to look into any added complexity for data updates when storing data in this manner.

Reply:
Based on your explanation, it makes sense now and it helps segregating sections as their properties may differ.